hedera-mirror-rest/controllers/contractController.js

```
@@ -656,6 +657,189 @@ class ContractController extends BaseController {
+    const {
+      bounds,
+      conditions: baseConditions,
+      params: baseParams,
+      timestampOrder,
+      indexOrder,
+      limit,
+    } = this.extractContractLogsQuery(filters, contractId);
+
+    const mapFilterKeyToColumn = {
+      [constants.filterKeys.INDEX]: constants.filterKeys.INDEX,
+      [constants.filterKeys.TIMESTAMP]: ContractLog.CONSENSUS_TIMESTAMP,
+    };
+    const indexBound = bounds[constants.filterKeys.INDEX];
+    const timestampBound = bounds[constants.filterKeys.TIMESTAMP];
+    let conditions = [];
+    const params = [...baseParams];
+
+    // if index is presented, at least 1 timestamp is required
+    if (!indexBound.isEmpty() && timestampBound.isEmpty()) {
+      throw new InvalidArgumentError('Cannot search by index without a timestamp');
+    }
+
+    if (timestampBound.hasBound() && timestampBound.hasEqual()) {
+      throw new InvalidArgumentError('Timestamp range must have gt (or gte) and lt (or lte)');
+    }
+
+    // validate index and timestamp operators compatibility
+    if (
+      indexBound.hasEqual() &&
+      timestampBound.hasLower() &&
+      timestampBound.lower.operator == utils.opsMap.gte &&
+      timestampBound.hasUpper() &&
+      timestampBound.upper.operator == utils.opsMap.lte
+    ) {
+      // invalid ops: index = & timestamp >= & timestamp <=
+      throw new InvalidArgumentError('Unsupported combination');
+    }
+
+    if (indexBound.hasEqual() && timestampBound.hasEqual() && timestampBound.hasBound()) {
+      // invalid ops: index = & timestamp = & timestamp >/>=/</<=
+      throw new InvalidArgumentError('Cannot support both range and equal');
+    }
+
+    if (
+      indexBound.hasLower() &&
+      !timestampBound.hasEqual() &&
+      (!timestampBound.hasLower() || timestampBound.lower.operator != utils.opsMap.gte)
+    ) {
+      // bounds should match index >/>= & timestamp =/>=
+      throw new InvalidArgumentError('Unsupported combination');
```

Explain what is not supported. It is hard to know from the message. Also, why this combination is not supported?

---

This is the case when `index` is `> or >=` and `timestamp` is `>`. If `timestamp` is `>` , `index` is not needed and instead of ignoring it, we are throwing an error. So, the `timestamp` should be `= or >=` to be a valid combination. I will update the error message.